src/components/chat/chat.js

```diff
@@ -1,12 +1,75 @@
 import React from 'react';
-import './chat.css';
+import './chat.scss';
 
 class Chat extends React.Component {
 
+  clickButton = (e) => {
+    e.preventDefault();
+    this.sendMessage();
+  }
+
+  detectEnter = (e) => {
+    var code = e.keyCode || e.which;
+    if (code === 13) {
+      e.preventDefault();
+      this.sendMessage();
+    }
+  }
+
+  sendMessage = () => {
+    const dateNow = new Date(Date.now());
+    const dateNowFormatted = dateNow.toLocaleString();
+    const msgToSend = {};
+    msgToSend.date = dateNowFormatted;
+    msgToSend.id = this.props.socket.id;
+    msgToSend.value = this.message.value;
+    this.props.socket.emit('chat message', msgToSend);
+    this.message.value = '';
+  }
+
+  componentDidMount() {
+    this.props.socket.on('chat message', (msg) => {
+      let currentId = this.props.socket.id;
+      let oneMessage = document.createElement('div');
+      let messages = document.getElementById('messages').appendChild(oneMessage);
+      oneMessage.className += ' oneMessage';
+      let messageText = document.createElement('div');
+      let messageDate = document.createElement('div');
+      oneMessage.appendChild(messageText);
+      messageText.innerHTML = msg.value;
+      oneMessage.appendChild(messageDate);
+      messageDate.innerHTML = msg.date;
+      messageDate.className += ' messageDate';
+      if (currentId === msg.id) messages.className += ' myMessage';
+      else messages.className += ' elseMessage';
+    });
+  }
+
   render() {
     return(
-      <div>
+      <div className="chat-component">
+        <div className="chat-header">
+          <div className="title">Question: How do I... blah blah blah</div>
+          <div className="hang-up">Hang Up</div>
+        </div>
+        <div className="chat-body">
+          <div className="editor">Code Editor</div>
+          <div className="chat-box">
```

Comment:
Would recommend adding replacing the styling sheet with the below to make chat scrollable in a window
```$background-white-color: white;
$background-grey-color: rgb(216, 216, 216);
$border-grey-color: rgb(150, 150, 150);

.chat-component {
  display: flex;
  flex-direction: column;
  height: 100vh;
  width: 100vw;
}

.chat-header {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
}

.title {
  background-color: $background-grey-color;
  margin: 20px 10px 20px 20px;
  padding: 20px;
  width: 80%;
  border: solid 1px $border-grey-color;
}

.hang-up {
  background-color: $background-grey-color;
  margin: 20px 20px 20px 10px;
  padding: 20px;
  border: solid 1px $border-grey-color;
  width: 100px;
  text-align: center;
}

.chat-body {
  display: flex;
  flex-direction: row;
  background-color: $background-white-color;
  height: 90vh;
}

.editor {
  width: 48%;
  background-color: $background-grey-color;
  margin: 20px 10px 20px 20px;
  border: solid 1px $border-grey-color;
}

.chat-box {
  width: 48%;
  background-color: $background-grey-color;
  margin: 20px 10px 20px 20px;
  border: solid 1px $border-grey-color;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  overflow: hidden;
}

form {
  padding: 5px 10px;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  min-height: 35px;
}

form input {
  width: 80%;
  height: 30px;
}

form button {
  width: 70px;
  margin-left: 5px;

}

form button:hover {
  cursor: pointer;
}

#messages {
  display: flex;
  flex-direction: column;
  overflow: auto;
  overflow-y: auto;
  vertical-align: bottom;
}

.oneMessage {
  display: flex;
  flex-direction: column;
  padding: 20px;
  margin: 5px 20px;
  border-radius: 10px;
  max-width: 50%;
}

.myMessage {
  align-self: flex-end;
  background-color: rgb(185, 249, 185);
}

.elseMessage {
  align-self: flex-start;
  background-color: white;
}

.messageDate {
  font-size: 10px;
  align-self: flex-end;
}

```